### PR TITLE
Feature/wimsr

### PR DESCRIPTION
**Pull request by nicholas-sly**

### CODE REVIEW

File: src/njoy21/input/WIMSR/src/readCard8.hpp [hunk 1]
```diff
@@ -0,0 +1,14 @@
+template< typename Istream >
+optional< Card8 > readCard8( Istream& is, optional< Card2a > card2a,
+                             const int jp1 ){
+  if( jp1 <= 0 ){
+    return std::nullopt;
+  } else if( card2a != std::nullopt ){
```
Comment (jlconlin) [inline src/njoy21/input/WIMSR/src/readCard8.hpp]: I don't think Card2a is required if `jp1>0`. You simply read in `jp1` values.

File: src/njoy21/input/WIMSR/test/WIMSR.test.cpp [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+        Card6b +
+        Card6c +
+        Card7 ) );
+
+      THEN( "an exception is thrown" ){
+        REQUIRE_THROWS( WIMSR( iss ) );
+      }
+    } // WHEN
+
+/*
```
Comment (jlconlin) [inline src/njoy21/input/WIMSR/test/WIMSR.test.cpp]: These comments should be removed.

File: src/njoy21/input/WIMSR/test/WIMSR.test.cpp [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+        REQUIRE( Approx( 0.6 ) == wimsr.card8->p1flx.value.at(0) );
+        REQUIRE( Approx( 0.7 ) == wimsr.card8->p1flx.value.at(1) );
+        REQUIRE( Approx( 0.8 ) == wimsr.card8->p1flx.value.at(2) );
+        REQUIRE( Approx( 0.9 ) == wimsr.card8->p1flx.value.at(3) );
+        REQUIRE( Approx( 1.0 ) == wimsr.card8->p1flx.value.at(4) );
+      } // THEN
+    } // WHEN
+  } // GIVEN
+
+  GIVEN( "invalid inputs" ){
```
Comment (jlconlin) [inline src/njoy21/input/WIMSR/test/WIMSR.test.cpp]: I really like how you considered all the different options.

File: src/njoy21/input/argument/parser/test/Base.test.cpp [hunk 4]
```diff
@@ -108,6 +108,22 @@ SCENARIO("Base"){
     }
   }
 
+  SECTION( "Vector of doubles" ){
```
Comment (jlconlin) [inline src/njoy21/input/argument/parser/test/Base.test.cpp]: 👍

File: src/njoy21/input/WIMSR/test/WIMSR.test.cpp (hunk 2 above)
Comment (nicholas-sly) [reply src/njoy21/input/WIMSR/test/WIMSR.test.cpp]: I keep wanting to check if extra values/cards are given, but I guess that's more in the scope of the next level up (input.hpp).  Will do.

File: src/njoy21/input/WIMSR/Card3/Rdfid.hpp [hunk 5]
```diff
@@ -5,7 +5,7 @@ struct Rdfid{
     return "The rdfid parameter is the identification number for the\n"
            "resonance data in the output of the WIMS library.";
   }
-  static bool verify( const Value_t rdfid ){
-    return ( rdfid > 0.0 and rdfid < 10000.0 );
+  static bool verify( Value_t rdfid, const int mat ){
+    return rdfid == double(mat);
```
Comment (jlconlin) [inline src/njoy21/input/WIMSR/Card3/Rdfid.hpp]: I don't believe these have to be equal. That being said, there isn't much information about what is an invalid `rdfid`. I think your previous verify was better.

**Pull request closed without merging**